HDFAttribIO compilation error on Summit

**Describe the bug**
Compiling on Summit I'm getting the compilation errors:

CMakeFiles/qmcwfs.dir/MuffinTin.cpp.o -c /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/MuffinTin.cpp
In file included from /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/MuffinTin.h:23,
                 from /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/EinsplineSet.h:25,
                 from /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/EinsplineSet.cpp:19:
/ccs/home/brtnfld/packages/qmcpack.master/src/Numerics/HDFNumericAttrib.h:88:8: error: redefinition of 'struct qmcplusplus::HDFAttribIO<long unsigned int>'
   88 | struct HDFAttribIO<unsigned long> : public HDFAttribIOBase
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~
/ccs/home/brtnfld/packages/qmcpack.master/src/Numerics/HDFNumericAttrib.h:63:8: note: previous definition of 'struct qmcplusplus::HDFAttribIO<long unsigned int>'
   63 | struct HDFAttribIO<hsize_t> : public HDFAttribIOBase
      |        ^~~~~~~~~~~~~~~~~~~~
In file included from /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/EinsplineSetBuilder.h:29,
                 from /ccs/home/brtnfld/packages/qmcpack.master/src/QMCWaveFunctions/EinsplineSetBuilderCommon.cpp:24:
/ccs/home/brtnfld/packages/qmcpack.master/src/Numerics/HDFNumericAttrib.h:88:8: error: redefinition of 'struct qmcplusplus::HDFAttribIO<long unsigned int>'
   88 | struct HDFAttribIO<unsigned long> : public HDFAttribIOBase
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~
/ccs/home/brtnfld/packages/qmcpack.master/src/Numerics/HDFNumericAttrib.h:63:8: note: previous definition of 'struct qmcplusplus::HDFAttribIO<long unsigned int>'
   63 | struct HDFAttribIO<hsize_t> : public HDFAttribIOBase
      |        ^~~~~~~~~~~~~~~~~~~~



**To Reproduce**
develop version of qmcpack

modules:
  1) lsf-tools/2.0   3) darshan-runtime/3.3.0-lite   5) DefApps      7) gcc/9.1.0      9) zlib/1.2.11          11) spectrum-mpi/10.4.0.3-20210112  13) fftw/3.3.9              15) python/3.8-anaconda3
  2) hsi/5.0.2.p5    4) xalt/1.2.1                   6) essl/6.3.0   8) cmake/3.21.3  10) netlib-lapack/3.9.1  12) boost/1.77.0                    14) netlib-scalapack/2.1.0

I'm using HDF5 1.13, built from source. Note that hsize_t is uint64_t.

the CMake command:

export HDF5_ROOT=$HOME/packages/hdf5-1.13/build/hdf5

cmake \
 -DQMC_MATH_VENDOR=IBM_MASS -DMASS_ROOT=/sw/summit/xl/16.1.1-10/xlmass/9.1.1 \
 -D CMAKE_BUILD_TYPE=Release \
 -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
 -D QMC_CUDA=0 \
 -D BUILD_LMYENGINE_INTERFACE=0 \
 -D MPIEXEC="jsrun" \
 ..

--- a/src/Numerics/HDFNumericAttrib.h
+++ b/src/Numerics/HDFNumericAttrib.h
@@ -57,33 +57,6 @@ struct HDFAttribIO<std::string>: public HDFAttribIOBase {
   }
   };*/
 
-
-/** Specialization for hsize_t */
-template<>
-struct HDFAttribIO<hsize_t> : public HDFAttribIOBase
-{
-  hsize_t& ref;
-
-  HDFAttribIO<hsize_t>(hsize_t& a) : ref(a) {}
-
-  inline void write(hid_t grp, const char* name) override
-  {
-    hsize_t dim     = 1;
-    hid_t dataspace = H5Screate_simple(1, &dim, NULL);
-    hid_t dataset   = H5Dcreate(grp, name, H5T_NATIVE_INT, dataspace, H5P_DEFAULT);
-    hid_t ret       = H5Dwrite(dataset, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, &ref);
-    H5Sclose(dataspace);
-    H5Dclose(dataset);
-  }
-
-  inline void read(hid_t grp, const char* name) override
-  {
-    hid_t h1  = H5Dopen(grp, name);
-    hid_t ret = H5Dread(h1, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, &ref);
-    H5Dclose(h1);
-  }
-};
-
 template<>
 struct HDFAttribIO<unsigned long> : public HDFAttribIOBase
 {

